File: python/005.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import fractions
from functools import reduce

def lcm(a, b): return int(a * b / fractions.gcd(a, b)) if a and b else 0

def p5(n):
    return reduce(lambda x, y:
                  lcm(x, y),
                  [ x for x in range(1, n + 1) ])

def main():
    print(p5(20))

if __name__ == '__main__':
    main()

File: elixir/005.exs
#!/usr/bin/env elixir

# https://projecteuler.net/problem=5

defmodule P5 do
	def gcd(a, 0) do a end
	def gcd(a, b) do gcd(b, rem(a,b)) end
	def lcm(a, b) do a * div(b, gcd(a, b)) end
end

1..20
|> Enum.reduce(fn(a, b) -> P5.lcm(a, b) end)
|> IO.inspect
